tests/test_utils.py

```diff
@@ -42,3 +45,91 @@ def test_numpy_round_trip(batch_size):
     np.testing.assert_allclose(
         R_np, R_recon, rtol=1e-5, atol=1e-6, err_msg=f"NumPy round-trip failed for batch size {batch_size}"
     )
+
+
+@pytest.mark.parametrize("batch_size", [0, 1, 100])
+def test_torch_identity_transform(batch_size):
+    if batch_size == 0:
+        pos = torch.randn(3)
+        b_pos = torch.randn(10, 3)
+    else:
+        pos = torch.randn(batch_size, 3)
+        b_pos = torch.randn(batch_size, 10, 3)
+
+    T_identity = torch.eye(4)
+
+    transformed = transform_by_T(pos, T_identity).cpu().numpy()
+    np.testing.assert_allclose(
```

Comment:
See my other review:
* No need to do `.numpy()` anywhere
* rather use our own `assert_allclose`
* `tolerance` should be a global constant after the imports.
* You need to jutify your tolerance threshold, why 1e-5 ? It is actually quite bad. And I possible, use `tol` instead of `atol` + `rtol`